File: src/day1/python/two_crystal_balls.py
from math import floor, sqrt


def two_crystal_balls(breaks: list[bool]) -> int:
    JMP = floor(sqrt(len(breaks)))
    idx = JMP

    # As long as the current element is false keep jumping (08/11/2023)
    while idx < len(breaks) and not breaks[idx]:
        idx += JMP

    idx -= JMP

    # +1 because we want to include the place we decremented from (09/11/2023)
    for i in range(idx, idx+JMP+1):
        if breaks[i]:
            return i
    return -1

File: src/day1/go/two_crystal_balls.go
package main

import "math"

func TwoCrystalBalls(breaks []bool) int {
	var jmp int = int(math.Sqrt(
		float64(len(breaks))))
	idx := jmp

	for ; idx < len(breaks); idx += jmp {
		if breaks[idx] {
			break
		}
	}

	idx -= jmp

	for ; idx < idx+jmp+1 && idx < len(breaks); idx++ {
		if breaks[idx] {
			return idx
		}
	}
	return -1
}
